code/Modules/Input/emsc/emscInputMgr.cc

```diff
@@ -82,6 +85,124 @@ emscInputMgr::discardCallbacks() {
     emscripten_set_touchcancel_callback("#canvas", 0, true, 0);
     emscripten_set_devicemotion_callback(0, true, 0);
     emscripten_set_deviceorientation_callback(0, true, 0);
+    emscripten_set_gamepaddisconnected_callback(this, true, emscGamepadDisconnected);
+}
+
+//------------------------------------------------------------------------------
+void
+emscInputMgr::updateGamepadsAttached(int& gamepadsAttached) {
+    gamepadsAttached = emscripten_get_num_gamepads();
+
+    // TODO: Check if gamepadsAttached <= MaxNumGamepads
+    int i = 0;
+    for (; i < 4; ++i) {
+        this->gamepad[i].attached = false;
+    }
+    for (i = 0; i < gamepadsAttached && i < 4; ++i) {
+        this->gamepad[i].attached = true;
+    }
+}
+
+//------------------------------------------------------------------------------
+void
+emscInputMgr::update() {
+    inputMgrBase::update();
+
+    int                     gamepadsAttached = 0;
+    uint32_t                down = 0;
+    uint32_t                up = 0;
+    uint32_t                pressed = 0;
+    EMSCRIPTEN_RESULT       result = EMSCRIPTEN_RESULT_FAILED;
+    EmscriptenGamepadEvent  gamepadState;
+
+    updateGamepadsAttached(gamepadsAttached);
+    for (int i = 0; i < gamepadsAttached; ++i) {
+        memset(&gamepadState, 0, sizeof(gamepadState));
```

Comment:
Please replace the memset() with Memory::Clear(&gamepadState, sizeof(gamepadState));